packages/@ember/-internals/glimmer/lib/utils/weak.ts

```diff
@@ -0,0 +1,86 @@
+import { Dict } from '@glimmer/interfaces';
+import { dict } from '@glimmer/util';
+
+const REFS = new WeakMap<object, WeakRef>();
+const VALS = new WeakMap<WeakRef, object>();
+
+export class WeakRef<T extends object = object> {
```

Comment:
Would you mind throwing some small local documentation for this? Obviously internal only, but it would be nice to have a summary of the intent here.